docs/cadl-for-openapi-dev.md

```diff
@@ -0,0 +1,285 @@
+
+| OpenAPI `response` field | Cadl construct                                     | Notes                    |
+| ------------------------ | -------------------------------------------------- | ------------------------ |
+| `description`            | `@doc` decorator                                   |                          |
+| `schema`                 | return type                                        |                          |
+| `headers`                | fields in the return type with `@header` decorator |                          |
+| `examples`               |                                                    | Not currently supported. |
+| `links` (OAS3)           |                                                    | Not currently supported. |
+
+## Schema Object
+
+OpenAPI schemas are represented in Cadl by [models](https://github.com/microsoft/cadl/blob/main/docs/tutorial.md#models).
+Models have any number of members and can extend and be composed with other models.
+
+Models can be defined with the `model` statement and then referenced by name, which generally results in a `$ref` to a schema for the model in the `definitions` or `components.schemas` section of the OpenAPI document.
+
+Cadl supports the ["spread" operator](https://github.com/microsoft/cadl/blob/main/docs/tutorial.md#spread) (`...`), which copies the members of the source model into the target model.
+But Cadl processes all spread transformations before emitters are invoked, so this form of reuse is not represented in the emitted OpenAPI.
+
+Cadl also supports single inheritance of models with the `extends` keyword. This construct can be used to produce an `allOf` with a single element (the parent schema) in OpenAPI. Cadl does not current provide a means to produce an `allOf` with more than one element -- these are generally treated as "composition" in code generators and thus better represented in Cadl with the spread operator.
+
+Cadl does not yet support a means to specify an OpenAPI `discriminator` but this support is currently in development.
+
+## definitions / components
+
+OpenAPI supports reuse of schemas, parameters, responses, and other elements with the `definitions` (OAS2) or `components` (OAS3) section of an OpenAPI definition.
+
+Referencing a model by name (not with "spread"), as an `op` parameter or return type or as the type of a property in another model, generally results in a `$ref` to a schema for the model in the `definitions` or `components.schemas` section of the OpenAPI document.
+
+Reusable parameters can be defined as members of a model and then incorporated into an operation parameter list using the spread operator. For example:
+
+```cadl
+model PetId {
+  @path petId: int32;
+}
+
+namespace Pets {
+  op read(... PetId): Pet | Error;
+}
+
+```
+
+results in a `$ref` to the named parameter `PetId` in either `parameters` or `components.parameters`.
+
+## Security Schemes Object
```

Comment:
missing not supported in this section maybe?

Reply:
These can be set through JavaScript so I added that.